flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -180,13 +190,28 @@ public void reconcile(FlinkResourceContext<CR> ctx) throws Exception {
         }
     }
 
-    private void applyAutoscaler(FlinkResourceContext<CR> ctx) throws Exception {
+    private void applyAutoscaler(FlinkResourceContext<CR> ctx, @Nullable String existingOverrides)
+            throws Exception {
         var autoScalerCtx = ctx.getJobAutoScalerContext();
         boolean autoscalerEnabled =
                 ctx.getResource().getSpec().getJob() != null
                         && ctx.getObserveConfig().getBoolean(AUTOSCALER_ENABLED);
         autoScalerCtx.getConfiguration().set(AUTOSCALER_ENABLED, autoscalerEnabled);
+
         autoscaler.scale(autoScalerCtx);
```

Comment:
Maybe add a comment explaining why this call to `scale` does not lead to the unnecessary update if we only patch the new config after it. I personally do not directly understand why this is the case.

Reply:
If I understand correct. The root cause is the logic related to `boolean specChanged` in `AbstractFlinkResourceReconciler#reconcile`. The `specChanged` will be true, when the ordering of the parallelism overrides is changed. And operator will scale job.

Right?

Reply:
Makes sense, thanks.